pkg/v1/tkg/managementcomponents/management_component_install.go

```diff
@@ -48,18 +61,207 @@ type ManagementComponentsInstallOptions struct {
+	addonSecret.Namespace = constants.TkgNamespace
+	log.Infof("Deleting %s/%s secret", addonSecret.Namespace, addonSecret.Name)
+	err := clusterClient.GetResource(addonSecret, addonSecret.Name, addonSecret.Namespace, nil, nil)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if controllerutil.ContainsFinalizer(addonSecret, addonFinalizer) {
+		controllerutil.RemoveFinalizer(addonSecret, addonFinalizer)
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		return clusterClient.UpdateResource(addonSecret, addonSecret.Name, addonSecret.Namespace)
+	})
+
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	err = clusterClient.DeleteResource(addonSecret)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete addon addonSecret %s", addonSecret)
+	}
+	return nil
+}
+
+func addonsPackageIsMovingRepositories(clusterClient clusterclient.Client, clusterName, addonName, namespace string) (bool, error) {
+	// We can check if the addons package has been installed from the core repository (https://gitlab.eng.vmware.com/core-build/tkg-packages/-/tree/main/core/addons-manager)
+	// by checking if the following secret exists.
+	addonSecret := &corev1.Secret{}
+	addonSecret.Name = generateAddonSecretName(clusterName, addonName)
+	addonSecret.Namespace = constants.TkgNamespace
+
+	err := clusterClient.GetResource(addonSecret, addonSecret.Name, addonSecret.Namespace, nil, nil)
+	if apierrors.IsNotFound(err) {
+		return false, err
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
 // InstallManagementComponents installs the management component to cluster
 func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error {
 	clusterClient, err := clusterclient.NewClient(mcip.ClusterOptions.Kubeconfig, mcip.ClusterOptions.Kubecontext, clusterclient.Options{})
 	if err != nil {
 		return errors.Wrap(err, "unable to get cluster client")
 	}
+	clusterName, err := clusterClient.GetCurrentClusterName(mcip.ClusterOptions.Kubecontext)
+	if err != nil {
+		return errors.Wrap(err, "unable to get cluster name")
+	}
+
+	// If the addons-manager is moving from core repository to management repository, its lifecycle management
+	// needs to be paused.
+	pauseAddonsManagerLifecycle, err := addonsPackageIsMovingRepositories(clusterClient, clusterName, addonsManagerName, constants.TkgNamespace)
+	if err != nil {
+		return err
+	}
+	if pauseAddonsManagerLifecycle {
+		err = PauseAddonLifecycleManagement(clusterClient, clusterName, addonsManagerName, constants.TkgNamespace)
+		if err != nil {
+			return err
+		}
+
+		err = NoopDeletePackageInstall(clusterClient, addonsManagerName, constants.TkgNamespace)
+		if err != nil {
+			return err
+		}
+
+		err = DeleteAddonSecret(clusterClient, clusterName, addonsManagerName, constants.TkgNamespace)
```

Comment:
we should delete the addons secret because if we do not the "legacy" controller will reinstall the old package.

Reply:
Even after pausing?

Reply:
... if we don't delete then we have to somehow make sure it never gets "unpaused"..

Reply:
Perhaps moving to InstallManagementPakcages will provide better impdepotency.

Reply:
Moved deleteaddonsecret until end of "installManagementPackage" 
manual testing showed the code is still idempotent.